pkg/services/rendering/http_mode.go

```diff
@@ -83,42 +112,85 @@ func (rs *RenderingService) renderViaHttp(ctx context.Context, renderKey string,
 		}
 	}()
 
+	var downloadFileName string
+	_, params, err := mime.ParseMediaType(resp.Header.Get("Content-Disposition"))
+	if err == nil {
```

Comment:
I don't understand why this is needed for CSV but not for rendering an image? If this fails the error is not handled and `downloadFileName` empty - what does that mean or what happens then?

Reply:
It could be added for images but it's not used now because images are not directly an attached file in emails so I thought we could handle it later if required.

If this fails, it's handled in the reporting function. It's not perfect but we don't have much information here to handle it properly. We could also use the generated unique filename but, then, it would not be possible to detect that it fails in the reporting function and use a better file name.

Reply:
So this is supposed to be the filename generated by Grafana UI when downloading CSV file?

I probably lack context, but if renderer fails to download the CSV file I would guess you get a non-200 response back from the service and can handle that as an error. In what circumstances are you expecting a file-payload returned but without a proper Content-Disposition header?

Reply:
Yes. And I agree, this should not happen. Do you think it would be better to just return the error and stop the function there?

Reply:
Yeah I would think returning the error would make more sense then.